frontend/src/App.tsx

```diff
@@ -43,6 +43,18 @@ const App = () => {
       }
       const data = await response.json();
       if (data) {
+        if (data.error != null) {
+          dispatch({
+            type: "SET_STATE",
+            state: {
+              linkToken: null,
+              linkErrorMessage: data.error.error_message,
+              linkErrorType: data.error.error_type,
+              linkErrorCode: data.error.error_code,
```

Comment:
These variable names are a bit confusing to me as this error is not a Link error (which is something else), it's a Link token API error right?. Maybe we could just add one key to the state that represents the actual error object and do something like this

```suggestion
              linkTokenError: data.error,
```

then later in the template 

```jsx

Error Code: <code>{linkTokenError.error_code}</code>

```

etc